tests/visualization_tests/test_slice.py

```diff
@@ -1,99 +1,373 @@
+        [create_study, None],
+        [_create_study_with_failed_trial, []],
+        [_create_study_with_failed_trial, ["param_a"]],
+        [_create_study_with_failed_trial, ["param_a", "param_b"]],
+        [_create_study_with_failed_trial, ["param_a", "param_b", "param_c"]],
+        [_create_study_with_failed_trial, ["param_a", "param_b", "param_c", "param_d"]],
+        [_create_study_with_failed_trial, None],
+        [prepare_study_with_trials, []],
+        [prepare_study_with_trials, ["param_a"]],
+        [prepare_study_with_trials, ["param_a", "param_b"]],
+        [prepare_study_with_trials, ["param_a", "param_b", "param_c"]],
+        [prepare_study_with_trials, ["param_a", "param_b", "param_c", "param_d"]],
+        [prepare_study_with_trials, None],
+        [_create_study_with_log_scale_and_str_category_2d, None],
+        [_create_study_mixture_category_types, None],
+    ],
+)
+def test_plot_slice(
+    plot_slice: Callable[..., Any],
+    specific_create_study: Callable[[], Study],
+    params: Optional[List[str]],
+) -> None:
+
+    study = specific_create_study()
+    figure = plot_slice(study, params=params)
+    if isinstance(figure, go.Figure):
+        figure.write_image(BytesIO())
+    else:
+        plt.savefig(BytesIO())
+
+
+def test_target_is_none_and_study_is_multi_obj() -> None:
+
+    study = create_study(directions=["minimize", "minimize"])
+    with pytest.raises(ValueError):
+        _get_slice_plot_info(study, None, None, "Objective Value")
+
+
+@pytest.mark.parametrize(
+    "specific_create_study",
+    [create_study, _create_study_with_failed_trial],
+)
+@pytest.mark.parametrize(
+    "params",
+    [
+        [],
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_get_slice_plot_info_empty(
+    specific_create_study: Callable[[], Study], params: Optional[List[str]]
+) -> None:
+
+    study = specific_create_study()
+    info = _get_slice_plot_info(study, params=params, target=None, target_name="Objective Value")
+    assert len(info.subplots) == 0
+
+
+def test_get_slice_plot_info_error() -> None:
+    study = prepare_study_with_trials()
+
+    with pytest.raises(ValueError):
+        _get_slice_plot_info(
+            study, params=["optuna", "Optuna"], target=None, target_name="Objective Value"
         )
+
+
+def test_get_slice_plot_info_non_exist_param_error() -> None:
+    study = prepare_study_with_trials()
+
+    with pytest.raises(ValueError):
+        _get_slice_plot_info(study, params=["optuna"], target=None, target_name="Objective Value")
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_get_slice_plot_info_params(params: Optional[List[str]]) -> None:
```

Comment:
This is the counterpart of 
- `test_get_contour_info_too_short_params`
- `test_get_contour_info_2_params`
- `test_get_contour_info_more_than_2_params`

in `test_contour.py`. Unlike `plot_contour`, plot slice's info can handle only one param case and the shape of sub info linearly depends on the size of params, so I unify them.

Reply:
For the same reason, this PR doesn't have the conterpart of `test_get_contour_info_log_scale_and_str_category_more_than_2_params`.